pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -89,7 +87,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	return nil
 }
 
-// GetDiscoveryClient ...
+// TODO: If we will invoke getDiscoveryClient multiple times in one reconcile stage, it's necessary move 'discovery' to the ReconcileInfinispan struct
```

Comment:
It would be good to add a short reason why

Reply:
yes, i will add that due to performance reason